concourse/scripts/self-update-pipeline.sh

```diff
@@ -23,7 +23,7 @@ else
   echo "Self update pipeline is enabled. Updating. (set SELF_UPDATE_PIPELINE=false to disable)"
 
   VAL_FROM_YAML=$(pwd)/paas-cf/concourse/scripts/val_from_yaml.rb
-  CONCOURSE_ATC_PASSWORD=$("$VAL_FROM_YAML" jobs.concourse.templates.atc.properties.basic_auth_password concourse-manifest/concourse-manifest.yml)
+  CONCOURSE_ATC_PASSWORD=$("$VAL_FROM_YAML" secrets.concourse_atc_password concourse-secrets/concourse-secrets.yml)
```

Comment:
Actually, all this PR can be about remove all the logic of retrieving the concourse password and only have it in the environments.sh